test/extended/psap/psap.go

```diff
@@ -0,0 +1,155 @@
+
+		//Save stalld startup log and compare the debug log after 10 minutes
+		//Extract sub-string of stalld PID from stalld log as [1] [1] [239426] [239426] [239426] [239426] [239426] [239426] [239426] [239426]
+		//Compare sub-string after 10 minutes, if no new pid or extra string found, it means stalld log no change, the stall didn't restart
+		exp := regexp.MustCompile(`\[.*?\]`)
+
+		stalldLogStdOut, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "journalctl", "-u", "stalld", "--since=@"+currentDatetime)
+		e2e.Logf("stalld log is:\n%s", stalldLogStdOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Matches would contain a slice of all successive matching text
+		// based on the regular expression.
+		allMatches := exp.FindAllString(stalldLogStdOut, -1)
+		previousMatches := strings.Join(allMatches, " ")
+		e2e.Logf("All matches in previous snapshot of stalld log < %v > ", previousMatches)
+
+		//Wait for 10 minutes and check stalld log in the meantime and exit if finding stalld restarted
+		errWait := wait.Poll(2*time.Minute, 10*time.Minute, func() (bool, error) {
+
+			stalldLogStdOut, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "journalctl", "-u", "stalld", "--since=@"+currentDatetime)
+			allMatches := exp.FindAllString(stalldLogStdOut, -1)
+			succedentMatches := strings.Join(allMatches, " ")
+			e2e.Logf("All matches in succedent snapshot of stalld log < %v > ", succedentMatches)
+
+			//the wait poll will exit if stalld log restart or anbonrmal
+			if err != nil || previousMatches != succedentMatches {
+				e2e.Logf("[ NOTE ] The stalld status abnormal or stalld service restarted:\n%v", stalldLogStdOut)
+				return true, nil
+			}
+			e2e.Logf("No restart of stalld process as expected, stalldLogStdOut is:\n%v", stalldLogStdOut)
+			return false, nil
+		})
+
+		if errWait != nil {
+			e2e.Logf("the error: %v", errWait)
+			return
+		}
+
+		err = fmt.Errorf("case: %v\nexpected error got because of %v", g.CurrentSpecReport().FullText(), fmt.Sprintf("stalld service restarted : %v", errWait))
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("[== Get stalld PID on labeled node ... ==]")
+		stalldPID, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "/bin/bash", "-c", "ps -efL| grep stalld | grep -v grep | awk '{print $2}'")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(stalldPID).NotTo(o.BeEmpty())
+
+		e2e.Logf("[== Get how many minutes stalld process keep up and running ... ==]")
+		stalldRuntimeMinSec, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "/bin/bash", "-c", "ps -o etime= -p "+stalldPID)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(stalldPID).NotTo(o.BeEmpty())
+		e2e.Logf("The the stalld process keep running for %v", stalldRuntimeMinSec)
+
+		//Double assert the execute time of stalld process
```

Comment:
Why  "double" assert?  Can we just not scrape the logs, wait for (re)start stalld service, wait a bit more than 600s and then check the pid again?

Reply:
Remove the assert, just keep how many minutes of stalld process running for logging.